document/core/binary/modules.rst

```diff
@@ -327,13 +328,14 @@ It decodes into a vector of :ref:`element segments <syntax-elem>` that represent
      \X{seg}^\ast{:}\Bsection_9(\Bvec(\Belem)) &\Rightarrow& \X{seg} \\
    \production{element segment} & \Belem &::=&
      \hex{00}~~e{:}\Bexpr~~y^\ast{:}\Bvec(\Bfuncidx)
-       &\Rightarrow& \{ \ETABLE~0, \EOFFSET~e, \EINIT~y^\ast \} \\
-   \production{element segment} & \Belem &::=&
-     \hex{01}~~y^\ast{:}\Bvec(\Bfuncidx)
-       &\Rightarrow& \{ \EINIT~y^\ast \} \\
-   \production{element segment} & \Belem &::=&
+       &\Rightarrow& \{ \ETABLE~0, \EOFFSET~e, \EINIT~((\REFFUNC~y)~\END)^\ast \} \\ &&|&
+     \hex{01}~~e^\ast{:}\Bvec(\Belemexpr)
+       &\Rightarrow& \{ \EINIT~e^\ast \} \\ &&|&
      \hex{02}~~x{:}\Btableidx~~e{:}\Bexpr~~y^\ast{:}\Bvec(\Bfuncidx)
-       &\Rightarrow& \{ \ETABLE~x, \EOFFSET~e, \EINIT~y^\ast \} \\
+       &\Rightarrow& \{ \ETABLE~x, \EOFFSET~e, \EINIT~((\REFFUNC~y)~\END)^\ast \} \\
+   \production{elemexpr} & \Belemexpr &::=&
+     \hex{D0} &\Rightarrow& \REFNULL~\END \\ &&|&
+     \hex{D2}~x{:}\Bfuncidx &\Rightarrow& (\REFFUNC~x)~\END \\
```

Comment:
Ah, I was rather referring to the end opcode (`\hex{0B}`) missing on the lhs. We don't need to represent it in the abstract syntax, but if you did you'd also need to adjust syntax/modules.rst.

Reply:
Oh, right. Well `\expr` uses `\END` in the syntax, so we may as well do that here until we merge this with the reference types proposal.